app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -0,0 +1,52 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun FavoriteButton(
+    isFavorite: Boolean,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit
+) {
+    Image(
+        painter = painterResource(
+            id =
+            if (!isFavorite) {
+                R.drawable.empty_heart
+            } else {
+                R.drawable.full_heart
+            }
+        ),
+        contentDescription = "Favorite button",
+        modifier = Modifier
+            .background(Color.Black.copy(0.7f), CircleShape)
+            .clip(CircleShape)
+            .clickable { onClick() }
+            .then(modifier)
```

Comment:
Passed `modifier` should be set first, and then all the other parameter. This is because we don't want to change the component itself internally, just its position, size, etc. For that to be possible we want to first include the changes made from the outside  and then changes made internally. 
You can read more about the order of modifiers here: [Order of modifiers matters](https://developer.android.com/jetpack/compose/modifiers#order-modifier-matters)